rascal-vscode-extension/src/dap/RascalDebugClient.ts

```diff
@@ -37,7 +37,8 @@ export class RascalDebugClient {
     debugSocketServersPorts: Map<number, number>; // Terminal processID -> socket server port for debug
     runningDebugSessionsPorts: Set<number>; // Stores all running debug session server ports
 
-
+    private portEventEmitter = new EventEmitter<undefined>();
```

Comment:
can we not give this a better type than `undefined`? something like` [number, number]` if we don't want to make a new class? or just inline anonymous class `{ processId: number; serverPort: number; }`?

Reply:
The debug view currently does not keep state regarding ports and process IDs (or, in fact, copy state from the debug client). When receiving an event, the debug view currently iterates over the open terminals and retrieves data from the debug client. In the current implementation, there is no real need to communicate the `processId:serverPort` pairs. Do you propose to keep a shadow bookkeeping in the view implementation, or do you prefer events having payload in general?

Reply:
I think for now, the fact that we don't keep state in the debug view is fine. I get why that would make the code harder.

So yes, I just prefer events to have a payload :) So that if later we want to use it, it's already there.

Reply:
Okay, fair enough. I agree.